components/layout_2020/flow/float.rs

```diff
@@ -84,66 +86,66 @@ pub(crate) struct PlacementAmongFloats<'a> {
     /// The next band, needed to know the height of the last band in current_bands.
     next_band: FloatBand,
     /// The size of the object to place.
-    object_size: LogicalVec2<Length>,
+    object_size: LogicalVec2<Au>,
     /// The minimum position in the block direction for the placement. Objects should not
     /// be placed before this point.
-    ceiling: Length,
+    ceiling: Au,
     /// The inline position where the object would be if there were no floats. The object
     /// can be placed after it due to floats, but not before it.
-    min_inline_start: Length,
+    min_inline_start: Au,
     /// The maximum inline position that the object can attain when avoiding floats.
-    max_inline_end: Length,
+    max_inline_end: Au,
 }
 
 impl<'a> PlacementAmongFloats<'a> {
     pub(crate) fn new(
         float_context: &'a FloatContext,
-        ceiling: Length,
-        object_size: LogicalVec2<Length>,
+        ceiling: Au,
+        object_size: LogicalVec2<Au>,
         pbm: &PaddingBorderMargin,
     ) -> Self {
-        assert!(!ceiling.px().is_infinite());
-        let mut current_band = float_context.bands.find(ceiling).unwrap();
-        current_band.top = ceiling;
+        assert!(ceiling.to_f32_px() != INFINITY);
+        let mut current_band = float_context.bands.find(ceiling.into()).unwrap();
+        current_band.top = ceiling.into();
         let current_bands = VecDeque::from([current_band]);
-        let next_band = float_context.bands.find_next(ceiling).unwrap();
+        let next_band = float_context.bands.find_next(ceiling.into()).unwrap();
         let min_inline_start = float_context.containing_block_info.inline_start +
-            pbm.margin.inline_start.auto_is(Length::zero);
+            pbm.margin.inline_start.auto_is(|| Au::zero().into()).into();
```

Comment:
I would keep `Length::zero` for now instead of `|| Au::zero().into()`

Reply:
why would you prefer to keep `Length::zero` for now? I am sure you have a reasoning behind it, so i have changed it to use `Length::zero` but i would also like to understand the reason.

Reply:
It seems strange to create `Au:zero()` just to immediately coerce it into `Length::zero()`.
Also, to reduce the diff in this patch. And once PaddingBorderMargin switches to Au, with `|| Au::zero().into()` the `into()` would become unnecessary but that can be hard to detect. With `Length::zero` the compiler will complain.